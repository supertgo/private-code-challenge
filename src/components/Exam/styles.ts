import styled from 'styled-components';

export const Wrapper = styled.div`
  width: 19rem;
  height: 19rem;
  display: flex;
  align-items: center; //centraliza horizontalmente
  justify-content: center; //cetraliza verticalmente
`;

export const Image = styled.img`
  width: 100%;
  max-width: 18rem;
`;
