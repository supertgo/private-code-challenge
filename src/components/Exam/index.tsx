import * as S from './styles';

export type ExamProps = {
  image: string;
};

const Exam = ({ image }: ExamProps) => (
  <S.Wrapper>
    <S.Image src={image} aria-label="exam image" />
  </S.Wrapper>
);

export default Exam;
