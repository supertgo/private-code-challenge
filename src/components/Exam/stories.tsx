import { Story, Meta } from '@storybook/react';
import Exam, { ExamProps } from '.';

export default {
  title: 'Exam',
  component: Exam
} as Meta;

export const Default: Story<ExamProps> = (args) => <Exam {...args} />;

Default.args = {
  image: '/img/enem.png'
};
