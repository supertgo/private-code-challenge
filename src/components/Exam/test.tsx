import { render, screen } from '@testing-library/react';

import Exam from '.';

const props = {
  image: '/img/enem.png'
};

describe('<Exam />', () => {
  it('should render the heading', () => {
    const { container } = render(<Exam {...props} />);

    expect(screen.getByRole('img', { name: /exam image/i })).toMatchSnapshot();

    expect(container.firstChild).toMatchSnapshot();
  });
});
