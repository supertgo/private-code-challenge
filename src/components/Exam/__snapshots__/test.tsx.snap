// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Exam /> should render the heading 1`] = `
.c0 {
  width: 100%;
  max-width: 18rem;
}

<img
  aria-label="exam image"
  class="c0"
  src="/img/enem.png"
/>
`;

exports[`<Exam /> should render the heading 2`] = `
.c0 {
  width: 19rem;
  height: 19rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 100%;
  max-width: 18rem;
}

<div
  class="c0"
>
  <img
    aria-label="exam image"
    class="c1"
    src="/img/enem.png"
  />
</div>
`;
